packages/next/src/client/components/app-router.tsx

```diff
@@ -204,6 +225,28 @@ function useNavigate(dispatch: React.Dispatch<ReducerActions>): RouterNavigate {
   )
 }
 
+const originalPushState =
+  typeof window !== 'undefined'
+    ? window.history.pushState.bind(window.history)
+    : null
+const originalReplaceState =
+  typeof window !== 'undefined'
+    ? window.history.replaceState.bind(window.history)
+    : null
+
+function copyNextJsInternalHistoryState(data: any) {
+  const currentState = window.history.state
+  const __NA = currentState?.__NA
+  if (__NA) {
+    data.__NA = __NA
```

Comment:
@timneutkens I believe `data` can be very plausible be `null` or `undefined` here based on the `replaceHistory` API.